### PR TITLE
Setup opt-in persistence for QueryClient

### PR DESCRIPTION
**Pull request by everdimension**

### CODE REVIEW

**Review by zerts — APPROVED** [review]

File: src/background/resource/sessionCacheService.ts [hunk 1]
```diff
@@ -1,11 +1,17 @@
+type Entry = { timestamp: number; value: unknown };
+
 export class SessionCacheService {
-  private cache: Record<string, { timestamp: number; value: unknown }> = {};
+  private cache: Record<string, Entry | undefined> = {};
```
Comment (vyorkin) [inline src/background/resource/sessionCacheService.ts:4]: Do we really need to store `undefined` values? It type checks successfully without `| undefined` btw

Comment (everdimension) [reply src/background/resource/sessionCacheService.ts:4]: That's the problem :) typescript assumes the value will always be retrieved by any key

Comment (everdimension) [reply src/background/resource/sessionCacheService.ts:4]: And we're not allowing to pass undefined when setting a value btw

Comment (vyorkin) [reply src/background/resource/sessionCacheService.ts:4]: ah, alright, makes sense!

**Review by vyorkin — APPROVED** [review]

**Pull request merged**